### DISCUSSION TITLE
Kata 5 Ejercicio 2

### DISCUSSION THREAD

**Discussion by Jorge-Macias-Cruz** [LaunchX-InnovaccionVirtual/CursoIntroPython#149] [Q&A] [answered]
tengo un error no se porque no me deja avanzar
[
![Captura de pantalla 2022-02-11 a la(s) 14 17 46](https://user-images.githubusercontent.com/37346944/153664202-c672ab0b-04b1-4426-a86d-22fa006ecc46.png)
](url)

**Comment by oscar-cbrlc** [comment, marked as answer]
Hola, creo que el error no es de código, si no de entrada. Cuando le diste a ejecutar ¿ingresaste el valor en la caja de texto que sale arriba? Lo digo porque me pasó lo mismo hasta que me di cuenta de eso :)

**Reply by abyleyva** [reply]
Este es por que no ingresaste valor en el input  (lo puedes ver donde dice..._with base 10: ' '_). Después de los dos puntos te muestra el valor que ingresaste, en este caso es un valor vacio

`[ValueError: invalid literal for int() with base 10: ' ']()`

En mi ejemplo, causa el error porque estoy agregando un valor flotante y lo quiero convertir a numero:
`[ValueError: invalid literal for int() with base 10: '3.1416']()`

Sucede con cualquier valor que ingreses en el input que no sea un valor entero, en este caso agregue un texto: _texto en lugar de numero_
`[ValueError: invalid literal for int() with base 10: 'texto en lugar de numero']()`

El error general se causa por que estamos convirtiendo lo que sea que se capture en el input a un valor tipo **int()**

Espero que te quede un poco mas claro.

**Comment by HeribertoBonilla** [comment]
Hola, también me sucedía, como dice Oscar se solucionó cuando al ejecute primero la función de leer los valores e ingresé los números en la caja de texto que sale arriba.

**Comment by Jorge-Macias-Cruz** [comment]
![Captura de pantalla 2022-02-11 a la(s) 14 42 06](https://user-images.githubusercontent.com/37346944/153667265-93bc4df2-e540-4e1f-b21d-5c7e6484ae0d.png)
Ahora tengo este problema

**Reply by oscar-cbrlc** [reply]
Tienes un typo. Escribiste 'planteta02' en lugar de 'planeta02' donde la convertiste a int

**Reply by abyleyva** [reply]
es como dice @OscarCabralesA tiene el error del typo entonces practicamente tienes tres variables:
**planeta01** convertido a int()
**planeta02** de tipo string (Recuerda que todo valor capturado de un input es  considerado un string
**planteta02** de tipo int(), creado por el error del typo

Entonces cuando quieres hacer una operación aritmética con **planeta01** tipo _int()_ y **planeta02** tipo _string_ son tipo de datos diferentes por eso es el error. Pero la raíz de todo es el typo al signar es algo que se tiene que tener cuidado o te dará muchos dolores de cabeza.

**Comment by jchz2** [comment]
Hola Amigos, por qué será que no me da el mismo resultado en la distancia de millas ?

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/71196434/153739170-8f542614-2c37-4d45-817e-fcdefa7bd451.png">

**Reply by jchz2** [reply]
Ya vi cuál era el detalle, gracias Bros!

**Comment by JoseAngelSilva2022** [comment]
> tengo un error no se porque no me deja avanzar [ ![Captura de pantalla 2022-02-11 a la(s) 14 17 46](https://user-images.githubusercontent.com/37346944/153664202-c672ab0b-04b1-4426-a86d-22fa006ecc46.png) ](url)

Hola amigo, espero que te encuentres bien, primero lo que quiero es preguntar si ingresaste un numero entero o decimal en tu primera ejecución, por lo que veo que tienes problema al convertirlo los datos de tipo Int.
trata de imprimir los datos que tienes para ver que ingresaste, pude ser que ahí venga el error

**Reply by jchz2** [reply]
Deben de ser los datos que tecleaste de entrada en tu primer input, yo lo tengo prácticamente igual y no me marca error

**Reply by Blackcat46** [reply]
Me paso lo mismo, se soluciono reiniciando el kernel. Solo cámbialo por otro y después lo cambias por el que tenias y se soluciona.

**Comment by gersondlarosa** [comment]
tu lo tienes a si 
planeta01 = int(planeta01)
planeta03 = int(planeta02)
--------------------------------------------------
y debes de declara cuanto vale la variable ya nos dice que son enteros no convertimos lo mejor es ponerlo a así y ya solo hacer la resta y la multiplicación 

planeta01 = 148597870
planeta03 = 778547200
